### ISSUE TITLE
inflight:This module is not supported, and leaks memory

### ISSUE THREAD

**Issue by sholtomaud** [aws/aws-cdk#32409]
### Describe the bug

npm warn deprecated inflight@1.0.6: This module is not supported, and leaks memory. Do not use it. Check out lru-cache if you want a good and tested way to coalesce async requests by a key value, which is much more comprehensive and powerful.
npm warn deprecated glob@7.2.3: Glob versions prior to v9 are no longer supported

### Regression Issue

- [ ] Select this option if this issue appears to be a regression.

### Expected Behavior

no deprecations or    leaks memory

### Current Behavior

deprecated     leaks memory

### Reproduction Steps

cdk init

### CDK CLI Version

2

### Node.js Version

20

### OS

macos

### Language

TypeScript

**Comment by pahud** [comment]
Can you share detailed info including

1. the full CDK version
2. the command and code snippets that reproduces this issue

Thank you.

**Comment by sholtomaud** [comment]
2.174.0 (build 9604329)

**Comment by chris-bateman** [comment]
Confirming same warning - 

``` bash 
npm warn deprecated inflight@1.0.6: This module is not supported, and leaks memory. Do not use it. Check out lru-cache if you want a good and tested way to coalesce async requests by a key value, which is much more comprehensive and powerful.
```

CDK 2.174.0 (build 9604329)
Node v22.12.0

**Comment by kaizencc** [comment]
@chris-bateman thanks for linking to the other issue. keeping this one closed as a duplicate of #31420. we'll look into it
